static/solo/js/index.js

```diff
@@ -51,24 +72,49 @@
       src: function(){
         var channel = s.channel,
             params = channel.parameters,
+            itemsPerPage = isset(s,'items_per_page') ? s.items_per_page : 8,
             url = '//api.tvpage.com/v1/channels/' + channel.id + '/videos?X-login-id=' + s.loginid;
 
         for (var p in params) { url += '&' + p + '=' + params[p];}
+        url += '&n=' + itemsPerPage + '&p=' + channelVideosPage;
         url += '&callback='+cbName;
         return url;
       }(),
       cbName: cbName
     },callback);
-  };
-
-  var render = function(idEl,target){
+    isFetching = false;
+  },
+  render = function(idEl,target){
     if (!idEl || !target) return console.log('need target');
     var frag = document.createDocumentFragment(),
     main = document.createElement('div');
     main.classList.add('tvp-player');
     main.innerHTML =  '<div id="tvp-player-el-'+idEl+'" class="tvp-player-el"></div></div>';
     frag.appendChild(main);
     target.appendChild(frag);
+  },
+  bindLoadMoreEvent = function(menu,data){
+    var scrollMenu = document.querySelectorAll('.ss-content')[0];
+      scrollMenu.addEventListener("scroll", debounce(function() {
+      var st = scrollMenu.scrollTop;
+      var nwh = document.body.clientHeight - scrollMenu.scrollHeight;
+      var percentDocument = (st*100)/nwh;
+      percentDocument = Math.round(percentDocument);
+      percentDocument = Math.abs(percentDocument);
+      if (percentDocument >= 55 && percentDocument <= 100) {
+        channelVideosPage++;
+        (function(unique,settings){
+          var menuSettings = JSON.parse(JSON.stringify(settings));
+          if (!lastPage) {
+            loadData(settings,unique,function(data){
+              lastPage = (!data.length || data.length < 0) ? true : false;
+              menuSettings.data = data || [];
+              menu.update(menuSettings,scrollMenu);
+            });
+          }
+        }(random(),getSettings('dynamic')));
```

Comment:
You can reduce all this code if you pass the settings to the `bindLoadMoreEvent` function as argument

```
if (playlistOption === 'show') {
    menuSettings.data = data || [];
    var menu = new Menu(player,menuSettings);
    bindLoadMoreEvent(menu, settings);
}
```